notes/2018-01-22-packaging.md

```diff
@@ -0,0 +1,104 @@
+Packaging
+=========
+
+This documents explains how we want to package and distribute Radicle.
+
+We will create packages for Brew, Debian, Pacman, and (possibly Nix). This
+packages will be made available through Github Releases.
+
+Haskell binaries will be statically linked with `-O3` optimization.
+
+Files to package
+----------------
+
+The Radicle package contains the `rad` entry command, the `rad-xxx` subcommands
+and Radicle source files.
+
+The `rad` command will be installed into a `PATH` directory. The subcommands and
+Radicle source files are installed into distribution specific directories. For
+Debian and Pacman this is `/usr/lib/radicle/modules` for the Radicle source
+files and `/usr/lib/radicle/bin` for the subcommands.
+
+Locating packaged files
+-----------------------
+
+The binaries distributed with the package need to locate files on the system the
+package is installed on.
+
+* The `rad` entry command locates all installed subcommands to be able to list
+  them and dispatch to them.
+* Radicle apps (implemented as subcommands) need to locate Radicle source files.
+* If Radicle app frontends that are written in radicle the interpreter also needs
+  to locate Radicle source files to be able to import them.
+
+In addition to installed packages the location mechanism needs to work for
+development builds (`stack build`), local tests, and CI tests.
+
+To implement file location of subcommands we are going to use Cabal’s
+[`Paths_radicle`][cabal-paths] module. `getBinDir` will return the folder
+containing all subcommands.
+
+File location for Radicle source files will be implemented by reading the
+`RADPATH` environment variable
+
+For local development and testing this mechanisms work out of the box. For
+packages the `rad` command will be a script with the following content.
+
+```
+#!/bin/sh
+radicle_bin=/usr/lib/radicle/bin
+RADPATH=/usr/lib/radicle/modules
+exec $radicle_bin/rad "@$"
+```
+
+Setting `radicle_bin` will determine the return value of `getBinDir`. Setting
```

Comment:
I believe this must be `<executable_name>_bindir`. Still, I think the default path should already point to the "right" directory (ie. build using `cabal`, which allows to tweak those via `cabal configure`).

Reply:
I checked and it is is `<package_name>_bindir`. For the second point see my answer above.